src/main/scala/com/databricks/spark/redshift/RedshiftJDBCWrapper.scala

```diff
@@ -267,7 +267,8 @@ private[redshift] class JDBCWrapper {
     // Somewhat hacky, but there isn't a good way to identify whether a table exists for all
     // SQL database systems, considering "table" could also include the database name.
     Try {
-      executeQueryInterruptibly(conn.prepareStatement(s"SELECT 1 FROM $table LIMIT 1")).next()
+      val stmt = conn.prepareStatement(s"SELECT 1 FROM $table LIMIT 1")
```

Comment:
If `$table` happens to contain an expensive-to-compute scalar subquery then this might be an expensive operation, so I've updated this to attempt to avoid execution using the same `.getMetaData()` trick.